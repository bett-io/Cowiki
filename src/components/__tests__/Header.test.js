/* eslint-disable react/jsx-filename-extension */

import React from 'react';

import toJson from 'enzyme-to-json';
import Header from '../Header';
import { shallow } from 'enzyme';

describe('Header component', () => {
  test('should pass snapshot check', () => {
    const rendered = shallow(<Header />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

  test('contains 3 <NavItem />s', () => {
    expect(shallow(<Header />).find('NavItem')).toHaveLength(3);
  });

  test('contains links to about, repos and hello', () => {
    const component = shallow(<Header />).find('LinkContainer');
    expect(component.findWhere(n => n.props().to === '/about')).toHaveLength(1);
    expect(component.findWhere(n => n.props().to === '/repos')).toHaveLength(1);
    expect(component.findWhere(n => n.props().to === '/repo')).toHaveLength(0);
    expect(component.findWhere(n => n.props().to === '/hello')).toHaveLength(1);
  });
});
