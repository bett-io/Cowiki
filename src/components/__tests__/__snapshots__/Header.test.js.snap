// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should pass snapshot check 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </NavbarBrand>
  </NavbarHeader>
  <Nav
    bsClass="nav"
    justified={false}
    pullLeft={false}
    pullRight={false}
    stacked={false}
  >
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/about"
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
      >
        About
      </NavItem>
    </LinkContainer>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/repos"
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
      >
        Repos
      </NavItem>
    </LinkContainer>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/hello"
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={3}
      >
        Hello
      </NavItem>
    </LinkContainer>
  </Nav>
  <Nav
    bsClass="nav"
    justified={false}
    pullLeft={false}
    pullRight={true}
    stacked={false}
  >
    <Connect(Login) />
  </Nav>
</Uncontrolled(Navbar)>
`;
